File: projects/order-creation-queries/repository/src/main/java/com/robinfood/repository/menu/IMenuRepository.java
package com.robinfood.repository.menu;

import com.robinfood.core.dtos.MenuCurrentDTO;
import com.robinfood.core.dtos.menu.MenuHallProductResponseDTO;
import com.robinfood.core.dtos.menu.MenuSuggestedPortionResponseDTO;
import com.robinfood.core.enums.Result;

import java.util.List;

/**
 * Repository for menu current related data
 */
public interface IMenuRepository {

    /**
     * @param brandId the brand to consult
     * @param countryId the country to consult
     * @param flowId the flow to consult
     * @param storeId the store to consult
     * @param token the authentication token
     * @return the menu current DTO
     */
    Result<MenuCurrentDTO> getMenuCurrent(
            Long brandId,
            Long countryId,
            Long flowId,
            Long storeId,
            String token
    );

    /**
     * Retrieves the product detail based on products identifiers given
     * @param token the authentication token to be used
     * @param countryId country identifier of the product
     * @param brandId brand identifier of the product
     * @param flowId flow identifier of the product
     * @param storeId store identifier of the product
     * @param productId the final product identifiers
     * @return the product with detailed info
     */
    Result<MenuHallProductResponseDTO> getProductDetail(
            Long brandId,
            Long countryId,
            Long flowId,
            Long productId,
            Long storeId,
            String token
    );

    /**
     * Retrieves the suggested portions detail based on portion identifiers given
     *
     * @param portionsIds the portions identifiers
     * @param token the authentication token to be used
     * @return the suggested portions with detailed info
     */
    Result<List<MenuSuggestedPortionResponseDTO>> getSuggestedPortions(
            List<Long> portionsIds,
            String token
    );
}

File: projects/posv2/repository/src/main/kotlin/com/robinfood/repository/menu/IMenuRepository.kt
package com.robinfood.repository.menu

import com.robinfood.core.dtos.menu.MenuBrandDTO
import com.robinfood.core.dtos.menu.MenuBrandResponseDTO
import com.robinfood.core.dtos.menu.response.MenuHallProductResponseDTO
import com.robinfood.core.dtos.menu.response.MenuSuggestedPortionResponseDTO
import com.robinfood.core.enums.Result

/**
 * Repository for menu related data
 */
interface IMenuRepository {

    /**
     * Retrieves the brands active for the store
     * [token] the authentication token to be used
     * [storeId] the store identification to get menu
     * @return the active menu for the store
     */
    suspend fun getStoreBrands(
            token: String,
            storeId: Long
    ): Result<List<MenuBrandResponseDTO>>

    /**
     * Retrieves the brand menu
     * [token] the authentication token to be used
     * [menuBrandResponse] the brand identification to get menu
     * [countryId] the store country identification
     * [flowId] flow identification
     * [storeId] the store identification to get menu
     * @return the active menu for the store
     */
    suspend fun getStoreMenuBrand(
            token: String,
            menuBrandResponse: MenuBrandResponseDTO,
            countryId: Long,
            flowId: Long,
            storeId: Long
    ): Result<MenuBrandDTO>

    /**
     * Retrieves the product detail based on products identifiers given
     * [token] the authentication token to be used
     * [countryId] country identifier of the product
     * [brandId] brand identifier of the product
     * [flowId] flow identifier of the product
     * [storeId] store identifier of the product
     * [platformId] platform identifier of the product
     * [productIds] the final product identifiers
     * @return the product with detailed info
     */
    suspend fun getProductDetail(
            token: String,
            countryId: Long,
            brandId: Long,
            flowId: Long,
            storeId: Long,
            platformId: Long?,
            productIds: Long
    ): Result<MenuHallProductResponseDTO>

    /**
     * Retrieves the suggested portions detail based on portion identifiers given
     * [token] the authentication token to be used
     * [portionsIds] the portions identifiers
     * @return the suggested portions with detailed info
     */
    suspend fun getSuggestedPortions(
            token: String,
            portionsIds: List<Long>
    ): Result<List<MenuSuggestedPortionResponseDTO>>
}